projects/packages/sync/src/modules/class-options.php

```diff
@@ -364,9 +364,14 @@ private function filter_theme_mods( &$value ) {
 	public function jetpack_sync_core_icon() {
 		$url = get_site_icon_url();
 
-		require_once JETPACK__PLUGIN_DIR . 'modules/site-icon/site-icon-functions.php';
+		$jetpack_url = \Jetpack_Options::get_option( 'site_icon_url' );
+		if ( function_exists( 'jetpack_site_icon_url' ) || defined( 'JETPACK__PLUGIN_DIR' ) ) {
```

Comment:
@zinigor - what do you think of this approach? Is this what you were looking for in [your comment](https://github.com/Automattic/jetpack/pull/18608#discussion_r606308780)? I'm surprised to see the `include_once` called even when the function already exists. But, since it's an `include_once`, it should be fine.

Reply:
the reason for the include_once was to not have another if clause, to distinguish the include. No harm in having it if we want to be explicit.

if( defined( 'JETPACK__PLUGIN_DIR' ) ) {
    if( ! function_exists( 'jetpack_site_icon_url' ) ) {
        include_once JETPACK__PLUGIN_DIR . 'modules/site-icon/site-icon-functions.php';
    }
    $jetpack_url = jetpack_site_icon_url();
}

Reply:
First of all, sorry for causing this fuss with the conditionals, but I really would like to see the code be more simple here. And I think Matthew's approach here aims to preserve the same logic that we had before we introduced the constant check.

So:

1. checking whether the constant is defined,
2. then checking whether the function is available - this check should be separate of the plugin check logic, because for some strange reason we could be in Jetpack context, but not have the file loaded;
3. then *requiring* the file as before, not including it (although here I don't know if it would matter that much - if the file is not there, the function call will still fatal after the `include`);

I hope that makes sense ^

Reply:
Done in 24106d2.